src/errors.rs

```diff
@@ -29,33 +29,40 @@ pub type Result<T> = std::result::Result<T, Error>;
 
 /// A code that specifies the PayError that occurred.
 #[derive(PartialEq, Eq, Debug, Clone)]
-#[repr(u8)]
 pub enum PayErrorCode {
+    /// An already recognized invoice tried to be paid.
+    /// Either a payment attempt is in progress or the invoice has already been paid.
+    /// There's no point in retrying this payment.
+    AlreadyUsedInvoice,
+
     /// The invoice has already expired.
-    /// There's no point in retrying this payment
+    /// There's no point in retrying this payment.
     InvoiceExpired,
-    /// An already recognized invoice tried to be paid. Either a payment attempt is in progress or the invoice has already been paid.
-    /// There's no point in retrying this payment
-    AlreadyUsedInvoice,
-    /// A locally issued invoice tried to be paid. Self-payments are not supported.
-    /// There's no point in retrying this payment
-    PayingToSelf,
+
     /// Not a single route was found.
-    /// There's no point in retrying this payment
+    /// There's no point in retrying this payment.
     NoRouteFound,
-    /// The recipient has rejected the payment.
-    /// It might make sense to retry the payment.
-    RecipientRejected,
-    /// Retry attempts or timeout was reached.
-    /// It might make sense to retry the payment.
-    RetriesExhausted,
-    /// All possible routes failed.
+
+    /// A locally issued invoice tried to be paid. Self-payments are not supported.
+    /// There's no point in retrying this payment.
+    PayingToSelf,
+
+    /// The payment failed. Likely the issue is on the receiver.
```

Comment:
Are we sure about this?

Reply:
Yes... The docs need to be improved. My interpretation of the code name is that this is a generic error, and not just for when the issue is the receiver. Some of these CLN errors seems like they could indicate a bug with our side (not 3L, but the payer implementation in general)